server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogReader.java

```diff
@@ -123,33 +130,44 @@ public void execute(String[] args) throws Exception {
       Set<Integer> tabletIds = new HashSet<>();
 
       for (String file : opts.files) {
-
         Path path = new Path(file);
         LogFileKey key = new LogFileKey();
         LogFileValue value = new LogFileValue();
 
-        if (fs.getFileStatus(path).isFile()) {
-          // read log entries from a simple hdfs file
-          try (final FSDataInputStream fsinput = fs.open(path);
-              DataInputStream input = DfsLogger.getDecryptingStream(fsinput, siteConfig)) {
-            while (true) {
-              try {
-                key.readFields(input);
-                value.readFields(input);
-              } catch (EOFException ex) {
-                break;
+        // read old style WALs
+        if (containsMapFile(fs, path)) {
```

Comment:
I think this check should happen in the else below for the temporary sorted WALs. The first `isFile()` check takes care of reading from the main WALs (the files in /accumulo/wal), which hasn't changed. I think the else below should read the files in /accumulo/recovery.

Reply:
That's how it was, but I ran into issues while trying to read the RFile with accumulo wal-info. It was simpler to figure out which version to use before figuring out if the user passed in a file or a directory.